docs/examples/creating-a-crud-api.md

```diff
@@ -0,0 +1,269 @@
+  },
+```
+
+Try sending `DELETE` to `http://localhost:8000/api/users/2` without a body.
+We'll get back:
+
+```
+user 2 deleted
+```
+
+## OPTIONS
+
+Options can be used for some advanced cases, including implementing preflight
+request checks for complex CORS use cases. See more on the
+[CORS documentation](/docs/examples/dealing-with-cors).
+
+## Full File Reference
+
+<details>
+<summary>[id].ts</summary>
+
+```ts
+import { Handlers } from "$fresh/server.ts";
+
+type User = {
+  id: string;
+  name: string;
+};
+
+const kv = await Deno.openKv();
+
+export const handler: Handlers<User | null> = {
+  async GET(_req, ctx) {
+    const id = ctx.params.id;
+    const user = await getUserById(id);
+    return new Response(JSON.stringify(user));
+  },
+  async DELETE(_req, ctx) {
+    const id = ctx.params.id;
+    await deleteUserById(id);
+    return new Response(`user ${id} deleted`);
+  },
+  async PUT(req, ctx) {
+    const id = ctx.params.id;
+    const body = await req.text();
+    const userBody = JSON.parse(body) as User;
+    updateUserById(id, userBody);
+    return new Response(JSON.stringify(userBody));
+  },
+};
+
+export async function getUserById(id: string): Promise<User> {
+  const key = ["user", id];
+  return (await kv.get<User>(key)).value!;
+}
+
+export async function deleteUserById(id: string) {
+  const userKey = ["user", id];
+  const userRes = await kv.get(userKey);
+  if (!userRes.value) return;
+
+  const ok = await kv.atomic()
+    .check(userRes)
+    .delete(userKey)
+    .commit();
+  if (!ok) throw new Error("Something went wrong.");
+}
+
+async function updateUserById(id: string, newUser: User) {
+  const userKey = ["user", id];
+  const userRes = await kv.get(userKey);
+  if (!userRes.value) return;
+
+  const ok = await kv.atomic()
+    .check(userRes)
+    .set(userKey, newUser)
+    .commit();
+  if (!ok) throw new Error("Something went wrong.");
+}
+```
+
+</details>
+
+<details>
+<summary>index.ts</summary>
+
+```ts
+import { Handlers } from "$fresh/server.ts";
+
+type User = {
+  id: string;
+  name: string;
+};
+
+const kv = await Deno.openKv();
+
+export const handler: Handlers<User | null> = {
+  async GET(_req, _ctx) {
+    const users = await getAllUsers();
+    return new Response(JSON.stringify(users));
+  },
+  async POST(req, _ctx) {
+    const body = await req.text();
+    const user = JSON.parse(body) as User;
+    await upsertUser(user);
+    return new Response(JSON.stringify(user));
+  },
+};
+
+export async function upsertUser(user: User) {
+  const userKey = ["user", user.id];
+
+  const oldUser = await kv.get<User>(userKey);
+
+  if (!oldUser.value) {
+    const ok = await kv.atomic()
+      .check(oldUser)
+      .set(userKey, user)
+      .commit();
+    if (!ok) throw new Error("Something went wrong.");
+  } else {
+    const ok = await kv.atomic()
+      .check(oldUser)
+      .set(userKey, user)
+      .commit();
+    if (!ok) throw new Error("Something went wrong.");
+  }
```

Comment:
Both branches are identical if I'm not mistaken.

Reply:
🧐  good catch. This is a result of not fully simplifying Andy's code, now that I'm not worried about email. Thanks.